torch/utils/_sympy/value_ranges.py

```diff
@@ -222,25 +212,23 @@ def eq(a, b):
     def ne(cls, a, b):
         return cls.not_(cls.eq(a, b))
 
-    @staticmethod
-    def lt(a, b):
+    @classmethod
+    def lt(cls, a, b):
         a = ValueRanges.wrap(a)
         b = ValueRanges.wrap(b)
-        if a.upper < b.lower:
-            return ValueRanges.wrap(sympy.true)
-        elif a.lower >= b.upper:
-            return ValueRanges.wrap(sympy.false)
-        return ValueRanges(sympy.false, sympy.true)
+        assert a.is_bool == b.is_bool
+        if a.is_bool:
+            return cls.and_(cls.not_(a), b)
```

Comment:
I find the following easier to follow than  `cls.and_(cls.not_(a), b)`:
```
if a.is_bool:
     a, b = (ValueRanges.increasing_map(val, lambda x: bool(x)) for val in (a, b))

if a.upper < b.lower:
...
```

Also I think `cls.and_(cls.not_(a), b)` gives wrong results for :


`False True < False True`.   `(True False) and (False True) -> False False` but we should be getting `False True`

Reply:
I don't think I follow your reasoning with the increasing map thing.

The thing here is that `a < b` for `a, b` bool is mathematically defined as `~a & b`. That's what we're using here.
Your example is wrong, Note that `not([False, True]) = [False, True]`. Your range is flipped. In fact, we check in the ValueRanges constructor that we never get ranges for which `lower > upper`.

Reply:
Shouldn't we be following python semantics here ? False < True -> True.